controllers/k6_start.go

```diff
@@ -15,6 +16,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func isPodReady(pod *v1.Pod) bool {
+	resp, err := http.Get(fmt.Sprintf("%v/v1/status", pod.Status.HostIP))
```

Comment:
It makes sense to add this check here:
```suggestion
        if len(pod.Status.HostIP) < 1 {
            return false
        }
	resp, err := http.Get(fmt.Sprintf("%v/v1/status", pod.Status.HostIP))
```
At first, this field is empty so there is no point in making any network requests then.